Update 2023-02-01-solidity-0.8.18-release-announcement.md (#206)
ethereum/solidity-blog: _posts/2023-02-01-solidity-0.8.18-release-announcement.md

We advise all Solidity developers to upgrade to version 0.8.18 to take advantage of these improvements and optimizations. Our team remains dedicated to continuously enhancing Solidity to benefit the entire Ethereum ecosystem.

Get ready to unleash your creativity and build impactful decentralized applications with the latest version of the Solidity Compiler. Happy coding!

> ## New linux binaries released for Solidity 0.8.18

Reviewer: I think we should make it clear this is an update to the original post (i.e. something you don't have to worry about if you did not update to 0.8.18 yet).

The text says that but I think that including something like `UPDATE` in the header would be clearer.

```suggestion
## UPDATE 2022-02-02: Rebuilt Linux binary for Solidity 0.8.18
```

The Linux binary for the 0.8.18 release was originally built against a too recent version of glibc and thereby did not run on older Linux systems. 

> Therefore, we rebuilt and replaced the binary so that you may see two different hashes for the ``solc-linux-amd64-v0.8.18+commit.87f61d96``. 

Reviewer: ```suggestion
Therefore, we rebuilt and replaced the binary so that you may see two different hashes for ``solc-linux-amd64-v0.8.18+commit.87f61d96``. 
```

> Therefore, we rebuilt and replaced the binary so that you may see two different hashes for the ``solc-linux-amd64-v0.8.18+commit.87f61d96``. 

Reviewer: "you may see two different hashes" sounds confusing to me. I'd clarify:
```suggestion
Therefore, we rebuilt and replaced the binary so that you may encounter an older version of ``solc-linux-amd64-v0.8.18+commit.87f61d96`` that does not match the hash you can find on the list of [currently published Linux binaries](binaries.soliditylang.org/linux-amd64/list.json).
```

> Therefore, we rebuilt and replaced the binary so that you may see two different hashes for the ``solc-linux-amd64-v0.8.18+commit.87f61d96``. 

Author: ```suggestion
Therefore, we rebuilt and replaced the binary so that you may encounter an older version of ``solc-linux-amd64-v0.8.18+commit.87f61d96`` that does not match the hash you can find on the list of [currently published Linux binaries](binaries.soliditylang.org/linux-amd64/list.json).
```

> Therefore, we rebuilt and replaced the binary so that you may see two different hashes for ``solc-linux-amd64-v0.8.18+commit.87f61d96``. 

Author: ```suggestion
Therefore, we rebuilt and replaced the binary so that you may encounter an older version of ``solc-linux-amd64-v0.8.18+commit.87f61d96`` that does not match the hash you can find on the list of [currently published Linux binaries](binaries.soliditylang.org/linux-amd64/list.json).
```

> The linux binary for the 0.8.18 release was built against a too recent version of glibc and does thereby not run on older linux systems. Therefore, we rebuilt and replaced the binary, so you may see two different hashes for the ``solc-linux-amd64-v0.8.18+commit.87f61d96``. Both versions of the compiler will behave identically other than the fact that the replaced one only runs on systems with glibc 2.32 or newer.

Author: ```suggestion
The Linux binary for the 0.8.18 release was built against a too recent version of glibc and does thereby not run on older Linux systems. 
Therefore, we rebuilt and replaced the binary so that you may see two different hashes for the ``solc-linux-amd64-v0.8.18+commit.87f61d96``. 
Both versions of the compiler will behave identically other than the fact that the replaced one only runs on systems with glibc 2.32 or newer.
```

> Therefore, we rebuilt and replaced the binary so that you may encounter an older version of ``solc-linux-amd64-v0.8.18+commit.87f61d96`` that does not match the hash you can find on the list of [currently published Linux binaries](binaries.soliditylang.org/linux-amd64/list.json).

Reviewer: The link is broken because I did not include the `https://` part:
```suggestion
Therefore, we rebuilt and replaced the binary so that you may encounter an older version of ``solc-linux-amd64-v0.8.18+commit.87f61d96`` that does not match the hash you can find on the [list of currently published Linux binaries](https://binaries.soliditylang.org/linux-amd64/list.json).
```

Both versions of the compiler will behave identically other than the fact that only the new one will work on systems with glibc versions older than 2.32.

To be clear: both binaries will produce identical outputs under all circumstances, including the commit hash in the metadata. Only the hash of the compiler binary itself will change due to the replacement, but the new binary will always produce byte-identical output.

In the listings at https://binaries.soliditylang.org/, the original, now replaced library, was listed as follows:

```
{
  "path": "solc-linux-amd64-v0.8.18+commit.87f61d96",
  "version": "0.8.18",
  "build": "commit.87f61d96",
  "longVersion": "0.8.18+commit.87f61d96",
  "keccak256": "0x265ad8479d7c5d30ef3556c5c23efddbf1718c1112abd219a6add420489cf930",
  "sha256": "0xa1c0f33eb4482c26f56719ecf62b0ee05d7d7a4f8264ffbddf9ebcd9095c32bd",
  "urls": [
    "bzzr://bc795017b74430520c9fd41300536f45d43edb74ca3c497ce2117d1954bbca12",
    "dweb:/ipfs/QmPkAoHf5hNoZS9W7uZhrGgVTSPoCui3h9MEEKTTfSe8am"
  ]
}

> ```

Author: ```suggestion
```

The new library hash is:

````
   {
      "path": "solc-linux-amd64-v0.8.18+commit.87f61d96",
      "version": "0.8.18",
      "build": "commit.87f61d96",
      "longVersion": "0.8.18+commit.87f61d96",
      "keccak256": "0x932901581a2ef0795851363522811fa160db5561e34cbb306e7105b6bbc3834d",
      "sha256": "0x95e6ed4949a63ad89afb443ecba1fb8302dd2860ee5e9baace3e674a0f48aa77",
      "urls": [
        "bzzr://61828ca55d703ddcd2ceff0e20a48ae6bca28a58b09725836893dc5f6b8e8018",
        "dweb:/ipfs/Qmf68UBPAspGAh6TJ1F9zKLFy3WjvH6W43HvzWWGAUQXTv"
      ]
    }
```
```

> ```
> {
>   "path": "solc-linux-amd64-v0.8.18+commit.87f61d96",
>   "version": "0.8.18",
>   "build": "commit.87f61d96",
>   "longVersion": "0.8.18+commit.87f61d96",
>   "keccak256": "0x265ad8479d7c5d30ef3556c5c23efddbf1718c1112abd219a6add420489cf930",
>   "sha256": "0xa1c0f33eb4482c26f56719ecf62b0ee05d7d7a4f8264ffbddf9ebcd9095c32bd",
>   "urls": [
>     "bzzr://bc795017b74430520c9fd41300536f45d43edb74ca3c497ce2117d1954bbca12",
>     "dweb:/ipfs/QmPkAoHf5hNoZS9W7uZhrGgVTSPoCui3h9MEEKTTfSe8am"
>   ]
> }
> ```

Reviewer: ````suggestion
In the listings at https://binaries.soliditylang.org, the original, now replaced library, was listed as follows:
```json
{
    "path": "solc-linux-amd64-v0.8.18+commit.87f61d96",
    "version": "0.8.18",
    "build": "commit.87f61d96",
    "longVersion": "0.8.18+commit.87f61d96",
    "keccak256": "0x265ad8479d7c5d30ef3556c5c23efddbf1718c1112abd219a6add420489cf930",
    "sha256": "0xa1c0f33eb4482c26f56719ecf62b0ee05d7d7a4f8264ffbddf9ebcd9095c32bd",
    "urls": [
        "bzzr://bc795017b74430520c9fd41300536f45d43edb74ca3c497ce2117d1954bbca12",
        "dweb:/ipfs/QmPkAoHf5hNoZS9W7uZhrGgVTSPoCui3h9MEEKTTfSe8am"
    ]
}
```
It has been replaced with:
```json
{
    "path": "solc-linux-amd64-v0.8.18+commit.87f61d96",
    "version": "0.8.18",
    "build": "commit.87f61d96",
    "longVersion": "0.8.18+commit.87f61d96",
    "keccak256": "0x932901581a2ef0795851363522811fa160db5561e34cbb306e7105b6bbc3834d",
    "sha256": "0x95e6ed4949a63ad89afb443ecba1fb8302dd2860ee5e9baace3e674a0f48aa77",
    "urls": [
        "bzzr://61828ca55d703ddcd2ceff0e20a48ae6bca28a58b09725836893dc5f6b8e8018",
        "dweb:/ipfs/Qmf68UBPAspGAh6TJ1F9zKLFy3WjvH6W43HvzWWGAUQXTv"
    ]
}
```
````

> ```

Author: ```suggestion
```

The new library hash is:

```
{
      "path": "solc-linux-amd64-v0.8.18+commit.87f61d96",
      "version": "0.8.18",
      "build": "commit.87f61d96",
      "longVersion": "0.8.18+commit.87f61d96",
      "keccak256": "0x932901581a2ef0795851363522811fa160db5561e34cbb306e7105b6bbc3834d",
      "sha256": "0x95e6ed4949a63ad89afb443ecba1fb8302dd2860ee5e9baace3e674a0f48aa77",
      "urls": [
        "bzzr://61828ca55d703ddcd2ceff0e20a48ae6bca28a58b09725836893dc5f6b8e8018",
        "dweb:/ipfs/Qmf68UBPAspGAh6TJ1F9zKLFy3WjvH6W43HvzWWGAUQXTv"
      ]
    }
```
```

We advise all Solidity developers to upgrade to version 0.8.18 to take advantage of these improvements and optimizations. Our team remains dedicated to continuously enhancing Solidity to benefit the entire Ethereum ecosystem.

Get ready to unleash your creativity and build impactful decentralized applications with the latest version of the Solidity Compiler. Happy coding!

## New linux binaries released for Solidity 0.8.18

> The Linux binary for the 0.8.18 release was built against a too recent version of glibc and does thereby not run on older Linux systems. 

Reviewer 2: ```suggestion
The Linux binary for the 0.8.18 release was originally built against a too recent version of glibc and thereby did not run on older Linux systems. 
```

Therefore, we rebuilt and replaced the binary so that you may see two different hashes for the ``solc-linux-amd64-v0.8.18+commit.87f61d96``. 

> Both versions of the compiler will behave identically other than the fact that the replaced one only runs on systems with glibc 2.32 or newer.

Reviewer 2: ```suggestion
Both versions of the compiler will behave identically other than the fact that only the new one will work on systems with glibc versions older than 2.32.
```

> Both versions of the compiler will behave identically other than the fact that only the new one will work on systems with glibc versions older than 2.32.

Reviewer 2: ```suggestion
Both versions of the compiler will behave identically other than the fact that only the new one will work on systems with glibc versions older than 2.32.

To be clear: both binaries will produce identical outputs under all circumstances, including the commit hash in the metadata. Only the hash of the compiler binary itself will change due to the replacement, but the new binary will always produce byte-identical output.
```

Since it came up in solc-tooling, maybe better to be crystal clear on this part?

Therefore, we rebuilt and replaced the binary so that you may see two different hashes for the ``solc-linux-amd64-v0.8.18+commit.87f61d96``. 
Both versions of the compiler will behave identically other than the fact that only the new one will work on systems with glibc versions older than 2.32.

To be clear: both binaries will produce identical outputs under all circumstances, including the commit hash in the metadata. Only the hash of the compiler binary itself will change due to the replacement, but the new binary will always produce byte-identical output.

In the listings at https://binaries.soliditylang.org/, the original, now replaced library, was listed as follows:

```
{
  "path": "solc-linux-amd64-v0.8.18+commit.87f61d96",
  "version": "0.8.18",
  "build": "commit.87f61d96",
  "longVersion": "0.8.18+commit.87f61d96",
  "keccak256": "0x265ad8479d7c5d30ef3556c5c23efddbf1718c1112abd219a6add420489cf930",
  "sha256": "0xa1c0f33eb4482c26f56719ecf62b0ee05d7d7a4f8264ffbddf9ebcd9095c32bd",
  "urls": [
    "bzzr://bc795017b74430520c9fd41300536f45d43edb74ca3c497ce2117d1954bbca12",
    "dweb:/ipfs/QmPkAoHf5hNoZS9W7uZhrGgVTSPoCui3h9MEEKTTfSe8am"
  ]
}
```

> 
> The new binary is listed as:
> 
> TBD

Author: ```suggestion
```

Both versions of the compiler will behave identically other than the fact that only the new one will work on systems with glibc versions older than 2.32.

To be clear: both binaries will produce identical outputs under all circumstances, including the commit hash in the metadata. Only the hash of the compiler binary itself will change due to the replacement, but the new binary will always produce byte-identical output.

In the listings at https://binaries.soliditylang.org, the original, now replaced binary, was listed as follows:
```json
{
    "path": "solc-linux-amd64-v0.8.18+commit.87f61d96",
    "version": "0.8.18",
    "build": "commit.87f61d96",
    "longVersion": "0.8.18+commit.87f61d96",
    "keccak256": "0x265ad8479d7c5d30ef3556c5c23efddbf1718c1112abd219a6add420489cf930",
    "sha256": "0xa1c0f33eb4482c26f56719ecf62b0ee05d7d7a4f8264ffbddf9ebcd9095c32bd",
    "urls": [
        "bzzr://bc795017b74430520c9fd41300536f45d43edb74ca3c497ce2117d1954bbca12",
        "dweb:/ipfs/QmPkAoHf5hNoZS9W7uZhrGgVTSPoCui3h9MEEKTTfSe8am"
    ]
}
```

> It has been replaced with:

Author: ```suggestion

It has been replaced with:
```

We advise all Solidity developers to upgrade to version 0.8.18 to take advantage of these improvements and optimizations. Our team remains dedicated to continuously enhancing Solidity to benefit the entire Ethereum ecosystem.

Get ready to unleash your creativity and build impactful decentralized applications with the latest version of the Solidity Compiler. Happy coding!

> ## UPDATE 2022-02-02: Rebuilt Linux binary for Solidity 0.8.18

Author: ```suggestion
## UPDATE 2023-02-02: Rebuilt Linux binary for Solidity 0.8.18
```

Get ready to unleash your creativity and build impactful decentralized applications with the latest version of the Solidity Compiler. Happy coding!

## UPDATE 2022-02-02: Rebuilt Linux binary for Solidity 0.8.18

The Linux binary for the 0.8.18 release was originally built against a too recent version of glibc and thereby did not run on older Linux systems. 
Therefore, we rebuilt and replaced the binary so that you may encounter an older version of ``solc-linux-amd64-v0.8.18+commit.87f61d96`` that does not match the hash you can find on the [list of currently published Linux binaries](https://binaries.soliditylang.org/linux-amd64/list.json).
Both versions of the compiler will behave identically other than the fact that only the new one will work on systems with glibc versions older than 2.32.

To be clear: both binaries will produce identical outputs under all circumstances, including the commit hash in the metadata. Only the hash of the compiler binary itself will change due to the replacement, but the new binary will always produce byte-identical output.

> In the listings at https://binaries.soliditylang.org/, the original, now replaced library, was listed as follows:
> 
> In the listings at https://binaries.soliditylang.org, the original, now replaced library, was listed as follows:

Reviewer: ```suggestion
In the listings at https://binaries.soliditylang.org, the original, now replaced binary, was listed as follows:
```
